app/uk/gov/hmrc/eusubsidycompliancefrontend/services/EscService.scala

```diff
@@ -55,18 +55,19 @@ class EscService @Inject() (escConnector: EscConnector)(implicit ec: ExecutionCo
       case Left(_) => Right(None)
       case Right(value) =>
         value.status match {
-          case NOT_FOUND => Right(None)
           case NOT_ACCEPTABLE =>
             value
               .parseJSON[UpstreamErrorResponse]
               .fold(_ => sys.error("Error in parsing Upstream Error Response"), Left(_))
           case OK =>
```

Comment:
Hmm....ok, looks like the backend is not returning a 404 anymore. It should in the case that the undertaking is not found. However, it shouldn't throw an exception when it returns a 404. I didn't spot this in your other PR.